src/Microsoft.TestPlatform.CoreUtilities/Utilities/StringExtensions.cs

```diff
@@ -17,5 +19,28 @@ public static string AddDoubleQuote(this string value)
         {
             return "\"" + value + "\"";
         }
+
+        public static void AppendToStringBuilderBasedOnMaxLength(this string data, StringBuilder result)
+        {
+            if (!string.IsNullOrEmpty(data))
+            {
+                // Don't append more data if already reached max length.
+                if (result.Length >= result.MaxCapacity)
+                {
+                    return;
+                }
+
+                // Add newline for readbility.
+                data += Environment.NewLine;
```

Comment:
If we are writing stringBuilder extension, then it should not be job of stringBuilder to add a new line. It should be the user's responsibility to add a new line.

Reply:
Create `StringBuilderExtensions` class with `AppendSafeWithNewLine()`.